Code/Framework/AzFramework/Tests/DeviceAttributeSystemComponentTests.cpp

```diff
@@ -0,0 +1,172 @@
+{
+    class DeviceAttributesSystemComponentTestFixture
+        : public LeakDetectionFixture
+    {
+    public:
+        DeviceAttributesSystemComponentTestFixture()
+            : LeakDetectionFixture()
+        {
+        }
+
+    protected:
+        void SetUp() override
+        {
+            m_settingsRegistry = AZStd::make_unique<AZ::SettingsRegistryImpl>();
+            AZ::SettingsRegistry::Register(m_settingsRegistry.get());
+
+            m_deviceAttributesSystemComponent = AZStd::make_unique<AzFramework::DeviceAttributesSystemComponent>();
+            m_deviceAttributesSystemComponent->Activate();
+        }
+
+        void TearDown() override
+        {
+            m_deviceAttributesSystemComponent->Deactivate();
+            m_deviceAttributesSystemComponent.reset();
+
+
+            AZ::SettingsRegistry::Unregister(m_settingsRegistry.get());
+            m_settingsRegistry.reset();
+        }
+
+        AZStd::unique_ptr<AzFramework::DeviceAttributesSystemComponent> m_deviceAttributesSystemComponent;
+        AZStd::unique_ptr<AZ::SettingsRegistryInterface> m_settingsRegistry;
+    };
+
+    class TestDeviceAttribute
+        : public AzFramework::DeviceAttribute
+    {
+    public:
+        using EvalFunc = AZStd::function<bool(AZStd::string_view)>;
+
+        TestDeviceAttribute(AZStd::string_view name, AZStd::string_view description,
+            AZStd::any value, EvalFunc eval)
+            : m_name(name)
+            , m_description(description)
+            , m_eval(eval)
+            , m_value(value)
+        {
+
+        }
+        ~TestDeviceAttribute() = default;
+
+        AZStd::string_view GetName() const override
+        {
+            return m_name;
+        }
+
+        AZStd::string_view GetDescription() const override
+        {
+            return m_description;
+        }
+
+        bool Evaluate(AZStd::string_view rule) const override
+        {
+            return m_eval(rule);
+        }
+
+        AZStd::any GetValue() const override
+        {
+            return m_value; 
+        }
+
+    private:
+        AZStd::string m_name;
+        AZStd::string m_description;
+        EvalFunc m_eval;
+        AZStd::any m_value;
+    };
+
+
+    TEST_F(DeviceAttributesSystemComponentTestFixture, DeviceAttributesSystem_Can_Register_Device_Attributes)
+    {
+        AZStd::string name { "TestAttribute" };
+        AZStd::string description { "Description" };
+        AZStd::string valueString{ "ABC123" };
+        AZStd::any value{ valueString };
+        TestDeviceAttribute::EvalFunc eval = [](AZStd::string_view rule)
+        {
+            // simple string comparison evaluator
+            return rule == "True";
+        };
+        auto deviceAttribute = AZStd::make_shared<TestDeviceAttribute>(name, description, value, eval);
+
+        // when the device attribute registrar exists
+        auto registrar = AzFramework::DeviceAttributeRegistrar::Get();
+        ASSERT_NE(registrar, nullptr);
```

Comment:
In Googletest you want the expected or unexpected value to be the first parameter and the actual value to the be the second parameter.
https://stackoverflow.com/questions/46873691/how-to-force-correct-order-of-expected-and-actual-in-google-test.

That way if the test fails it will output that nullptr is "not expected"